### PR TITLE
Revert "Revert "stats: update method names of counter"

### PR DESCRIPTION
**Pull request by jingwei99**
Reverts lyft/envoy-mobile#1153

Description: Update the method names of counters to follow the same convention as Gauge. This PR also added integration tests for counter and gauge to guard related changes from crashing on the platform side.
Risk Level: Moderate
Testing: Local and CI

Signed-off-by: Jingwei Hao jingwei.hao@gmail.com

### CODE REVIEW

File: examples/java/hello_world/MainActivity.java [hunk 1]
```diff
@@ -127,4 +131,19 @@ private void makeRequest() {
...
+
+    final Gauge gauge =
+        engine.statsClient().gauge(new Element("foo"), new Element("bar"), new Element("gauge"));
+
+    counter.increment(1);
+    counter.increment(5);
+
+    gauge.set(5);
+    gauge.add(5);
+    gauge.sub(5);
```
Comment (rebello95) [inline examples/java/hello_world/MainActivity.java:147]: Maybe vary these numbers?

File: examples/java/hello_world/hello_world.iml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (rebello95) [inline examples/java/hello_world/hello_world.iml:1]: @buildbreaker is this file required?

File: examples/objective-c/hello_world/ViewController.m [hunk 3]
```diff
@@ -123,6 +126,20 @@ - (void)addResponseMessage:(NSString *)message
   [self.tableView reloadData];
 }
 
+- (void)sendStats {
+  Element *eleFoo = [[Element alloc] initWithStringLiteral:@"foo"];
```
Comment (rebello95) [inline examples/objective-c/hello_world/ViewController.m:130]: Please spell out "element" (more idiomatic on iOS)

File: examples/objective-c/hello_world/ViewController.m [hunk 4]
```diff
@@ -123,6 +126,20 @@ - (void)addResponseMessage:(NSString *)message
...
+  Element *eleBar = [[Element alloc] initWithStringLiteral:@"bar"];
+  Element *eleCounter = [[Element alloc] initWithStringLiteral:@"counter"];
+  Element *eleGauge = [[Element alloc] initWithStringLiteral:@"gauge"];
+  id<Counter> counter = [self.statsClient counterWithElements:@[ eleFoo, eleBar, eleCounter ]];
+  [counter incrementWithCount:1];
+
+  id<Gauge> gauge = [self.statsClient gaugeWithElements:@[ eleFoo, eleBar, eleGauge ]];
+  [gauge setWithValue:1];
+  [gauge addWithAmount:1];
+  [gauge subWithAmount:1];
```
Comment (rebello95) [inline examples/objective-c/hello_world/ViewController.m:140]: Why are these different from Android?

**Review by rebello95 — COMMENTED** [review]
LGTM overall, couple small nits and one question

**Comment by rebello95** [comment]
Also what ref of Envoy are you bumping to? Doesn't look like the `main` branch?

File: examples/java/hello_world/hello_world.iml (hunk 2 above)
Comment (jingwei99) [reply examples/java/hello_world/hello_world.iml:1]: my ide generated it, it's not required

File: examples/objective-c/hello_world/ViewController.m (hunk 4 above)
Comment (jingwei99) [reply examples/objective-c/hello_world/ViewController.m:140]: you mean the method names? 

they come from the swift generated `Envoy/Envoy-Swift.h`

File: examples/java/hello_world/MainActivity.java (hunk 1 above)
Comment (jingwei99) [reply examples/java/hello_world/MainActivity.java:147]: > Also what ref of Envoy are you bumping to? Doesn't look like the `main` branch?

my local branch is pointed to a different version of envoy, will revert

File: examples/objective-c/hello_world/ViewController.m (hunk 4 above)
Comment (rebello95) [reply examples/objective-c/hello_world/ViewController.m:140]: I meant the amounts

**Comment by jingwei99** [comment]
> Also what ref of Envoy are you bumping to? Doesn't look like the `main` branch?

my local branch is pointed to a different version of envoy, will revert

File: examples/objective-c/hello_world/ViewController.m (hunk 4 above)
Comment (jingwei99) [reply examples/objective-c/hello_world/ViewController.m:140]: to clarify: you mean the actual numbers sent (instead of the method names)

**Review by rebello95 — APPROVED** [review]

File: examples/objective-c/hello_world/ViewController.m [hunk 5]
```diff
@@ -46,8 +47,10 @@ - (void)startEnvoy {
     NSLog(@"starting Envoy failed: %@", error);
   } else {
     NSLog(@"started Envoy, beginning requests...");
-    self.client = [engine streamClient];
+    self.streamClient = [engine streamClient];
+    self.statsClient = [engine statsClient];
     [self startRequests];
+    [self sendStats];
```
Comment (rebello95) [inline examples/objective-c/hello_world/ViewController.m:53]: I just noticed this - it looks like iOS is sending stats only once whereas Android is sending them with each request, is that correct? If so, these should be consistent

Comment (jingwei99) [reply examples/objective-c/hello_world/ViewController.m:53]: looking, the structure of the code for ios and android are pretty different

**Review by rebello95 — APPROVED** [review]
